### PR TITLE
[new feature] add post edit button

### PR DESCRIPTION
**Pull request by xu-song**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the new behavior?

Add `edit` button for each post.

It provides a way of `online edit`. In addition, links to github is a good way for`collaborative edit`.

* Link to demo site with this changes:  https://blog.eson.org/pub/92bb368/

### How it looks?
**common style**

**hover style**

Live demo [here](https://blog.eson.org/pub/92bb368/).

### How to use?
In NexT `_config.yml`:

```yml
# Post edit
post_edit:
  enable: true
  url: https://github.com/ESblog/_posts/tree/master/   # view source
# url: https://github.com/ESblog/_posts/edit/master/   # fork and edit
```
The above demo page, links to 
`https://github.com/ESblog/_posts/tree/master/web/blog-framework/nodejs-hexo/plugin/hexo-plugin-recommendation.md` according to the file path`_post/web/blog-framework/nodejs-hexo/plugin/hexo-plugin-recommendation.md`.

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

File: layout/_macro/post.swig [hunk 1]
```diff
@@ -59,7 +59,12 @@
                 <a class="post-title-link" href="{{ url_for(post.path) }}" itemprop="url">{#
                 #}{{ post.title | default(__('post.untitled'))}}{#
               #}</a>{#
-            #}{% else %}{{ post.title }}{% endif %}{#
+            #}{% else %}
+              {{ post.title }}
+              {% if theme.post_edit.enable %} 
+                <a href="{{theme.post_edit.url}}{{post.source.slice(7)}}" class="article-edit-link" title="{{ __('post.edit') }}"><i class="fa fa-pencil"></i></a>
```
Comment (ivan-nginx) [inline layout/_macro/post.swig]: Can i ask what's mean `post.source.slice(7)`? I'm interesting in `slice(7)` parameter.

Comment (xu-song) [reply layout/_macro/post.swig]: in the above example
```
post.source = `_post/web/blog-framework/nodejs-hexo/plugin/hexo-plugin-recommendation.md`
```

`post.source.slice(7) = web/blog-framework/nodejs-hexo/plugin/hexo-plugin-recommendation.md`

**Comment by xu-song** [comment]
# URL setting
url setting depends on the github project.

# If git rep

File: layout/_macro/post.swig (hunk 1 above)
Comment (xu-song) [reply layout/_macro/post.swig]: I am not sure whether all the `post.source` starts with `_post/`. Maybe need to be optimized.

**Comment by ivan-nginx** [comment]
Ooh, i see. It's cutted out main Github link of repository, right?

**Comment by xu-song** [comment]
Yes.  Any suggestion and optimization is welcome.

**Comment by sli1989** [comment]
Need upload resource md?

**Comment by xu-song** [comment]
Yes.
If you needs collaborative editing, like `https://hexo.io/docs/`, you need to create a source repository.

## site repository
e.g. 
- https://github.com/hexojs/site/
- https://github.com/hexojs/hexo-starter

##  post only
e.g.
- https://github.com/ESblog/_posts

Resource md is an repository.

**Pull request merged**